conda_smithy/feedstock_content/ci_support/checkout_code.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+
+# Update PR refs for testing.
+if [[ -n "${CIRCLE_PR_NUMBER}" ]]
+then
+    FETCH_REFS="${FETCH_REFS} +refs/pull/${CIRCLE_PR_NUMBER}/head:pr/${CIRCLE_PR_NUMBER}/head"
+    FETCH_REFS="${FETCH_REFS} +refs/pull/${CIRCLE_PR_NUMBER}/merge:pr/${CIRCLE_PR_NUMBER}/merge"
+fi
+
+# Retrieve the refs.
+git fetch -u origin ${FETCH_REFS}
+
+# Checkout the PR merge ref.
+if [[ -n "${CIRCLE_PR_NUMBER}" ]]
+then
+    git checkout -qf "pr/${CIRCLE_PR_NUMBER}/merge"
+fi
+
+# Check for merge conflicts.
+if [[ -n "${CIRCLE_PR_NUMBER}" ]]
+then
+    git checkout -qf "pr/${CIRCLE_PR_NUMBER}/merge"
+    git branch --merged | grep "pr/${CIRCLE_PR_NUMBER}/head" > /dev/null
```

Comment:
Ideally we would also check for merge conflicts and error out if present. This makes some effort at it. However, we can't really do that well without knowing what the base branch is. As it may not be `master`, we really need CircleCI to provide that [information]( https://discuss.circleci.com/t/add-environment-varible-circle-pr-base-branch/6189 ). Then we could do something like [this]( https://github.com/conda-forge/staged-recipes/pull/1486/files#diff-f9e6910612efc2dbe7c3774b355f2d97R27 ).